scipy/optimize/_differentialevolution.py

```diff
@@ -1363,3 +1522,29 @@ def violation(self, x):
         excess_ub = np.maximum(ev - self.bounds[1], 0)
 
         return excess_lb + excess_ub
+
+
+def rank(feasible, energies, constraint_violations):
+    # rank population members in order of fitness
+    # feasible : boolean array (M,) where M is the number of population
+    #   members
+    # energies : double array (M,)
+    # constraint_violations : double array (M, P) where P is the number of
+    #   constraints
+    M = np.size(energies)
```

Comment:
Could already use `np.arange(np.size(energies))` to not repeat.

But you can use `np.where(feasible)` (and `np.where(~feasible)`) instead of `np.arange(M)[feasible]`. Cleaner IMO.